tlsfuzzer/expect.py

```diff
@@ -179,6 +179,8 @@ def process(self, state, msg):
                     self.extensions[ext_id](state, ext)
                 if ext_id == ExtensionType.extended_master_secret:
                     state.extended_master_secret = True
+                if ext_id == ExtensionType.encrypt_then_mac:
+                    state.encrypt_then_mac = True
```

Comment:
Section 3.1 in RFC 7366 states that while upgrade to EtM is possible, Section 2 states that "client and server MUST NOT use encrypt-then-MAC unless both sides have successfully exchanged encrypt_then_mac extensions".

So, either the state of the `encrypt_then_mac` needs to be reset to `False` on every renegotiation/resumption and that logic needs to be implemented in test scripts, or we need to put that logic into the fuzzer itself